state-chain/pallets/cf-trading-strategy/src/lib.rs

```diff
@@ -154,8 +159,11 @@ pub mod pallet {
 		fn on_idle(_current_block: BlockNumberFor<T>, remaining_weight: Weight) -> Weight {
 			let mut weight_used: Weight = T::DbWeight::get().reads(1);
 
-			// TODO: use safe mode here checking if auto strategies are enabled
+			if !T::SafeMode::get().strategy_execution_enabled {
+				return weight_used
+			}
 
+			weight_used += T::DbWeight::get().reads(1);
```

Comment:
FWIW I think we can assume the read access for safe mode is effectively free: because it's read all over the place, it is likely to be in the storage cache. 

(This is what is meant by 'whiltelisted storage' in the benchmarks: storage that is frequently accessed and therefore (almost) guaranteed to be cached.)